### PR TITLE
README: Add examples for "makes a new array" versus "write into an existing one"

### PR DESCRIPTION
**Pull request by DilumAluthge**
This was an early stumbling block for me (see e.g. https://discourse.julialang.org/t/tullio-overwrite-existing-matrix-instead-of-creating-a-new-matrix/51834).

I had trouble finding this in the README. I figured it might be easier to find if we added some examples.

### CODE REVIEW

**Comment by DilumAluthge** [comment]
I think this is a little too wordy and makes this bullet point harder to read. Maybe there's a different place to put these examples.

**Comment by mcabbott** [comment]
The readme isn't great, I'm sorry. It's helpful to know where things are confusing. I've been fiddling with a longer one but am not so happy with it, not sure how to get the right balance... I have however added to the readme another block of simpler examples, for now.
